lib/compress.js

```diff
@@ -1749,7 +1749,7 @@ merge(Compressor.prototype, {
                     extract_declarations_from_unreachable_code(compressor, self.body, a);
                     return make_node(AST_BlockStatement, self, { body: a });
                 }
-            } else {
+            } else { // self instanceof AST_Do
```

Comment:
I could do `} else /*if (self instanceof AST_Do)*/ {` if that pleases the gods... 😉